.github/workflows/haskell.yml

```diff
@@ -31,214 +30,158 @@ jobs:
+          BOOTSTRAP_HASKELL_ADJUST_CABAL_CONFIG=1 \
+          BOOTSTRAP_HASKELL_ADJUST_BASHRC=1 \
+          BOOTSTRAP_HASKELL_GHC_VERSION="${{ matrix.ghc }}" \
+          BOOTSTRAP_HASKELL_CABAL_VERSION="3.6.2.0" \
+          sh
 
-    - name: Add build script path
-      run: echo "$(pwd)/.github/bin" >> $GITHUB_PATH
 
-    - name: Install pkgconfiglite
-      if: matrix.os == 'windows-latest'
-      run: retry 2 choco install -y pkgconfiglite
+        cat <(echo "source /c/ghcup/env") ~/.bashrc > ~/.bashrc.new
+        mv ~/.bashrc.new ~/.bashrc
 
-    - name: Install libsodium (Windows)
-      if: matrix.os == 'windows-latest'
-      run: |
-        curl -Ls \
-          --connect-timeout 5 \
-          --max-time 10 \
-          --retry 5 \
-          --retry-delay 0 \
-          --retry-max-time 40 \
-          https://download.libsodium.org/libsodium/releases/libsodium-1.0.18-mingw.tar.gz -o libsodium-1.0.18-mingw.tar.gz
-        tar zxvf libsodium-1.0.18-mingw.tar.gz
+        source ~/.bashrc
 
-        sed -i "s|/d/a/1/s/|D:/a/cardano-node/cardano-node/|g" libsodium-win64/lib/pkgconfig/libsodium.pc
+        echo "# Original cabal config extra-lib-dirs"
+        grep extra-lib-dirs /c/cabal/config
 
-        export PKG_CONFIG_PATH="$PKG_CONFIG_PATH;$(readlink -f libsodium-win64/lib/pkgconfig | sed 's|^/d|D:|g' | tr / '\\')"
-        echo "PKG_CONFIG_PATH=$PKG_CONFIG_PATH" >> $GITHUB_ENV
+        sed -i 's/C:\\msys64\\mingw64\\lib/C:\\ghcup\\ghc\\8.10.7\\mingw\\x86_64-w64-mingw32\\lib, C:\\msys64\\mingw64\\lib/g' /c/cabal/config
 
-        export LIBSODIUM_PATH="$(readlink -f libsodium-win64/bin | sed 's|^/d|D:|g' | tr / '\\')"
-        echo "LIBSODIUM_PATH=$LIBSODIUM_PATH"
-        echo "$LIBSODIUM_PATH" >> $GITHUB_PATH
+        echo "# Modified cabal config extra-lib-dirs"
+        grep extra-lib-dirs /c/cabal/config
 
-        echo "# pkg-config libsodium"
-        pkg-config libsodium --cflags --libs
+        ghc --version
+        cabal --version
 
-    - name: Install libsodium (MacOS)
-      if: matrix.os == 'macos-latest'
-      run: brew install libsodium
+    - name: Set cache version
+      run: echo "CACHE_VERSION=grFfw8r" >> $GITHUB_ENV
 
-    - name: Setup Github Scripts
-      run: echo "$(pwd)/.github/bin" >> $GITHUB_PATH
+    - uses: actions/checkout@v2
 
-    - name: Haskell versions
+    - name: Add build script path
       run: |
-        ghc --version
-        cabal --version
+        if [ "${{ runner.os }}" == "Windows" ]; then
+          # MSYS2 doesn't inherit $GITHUB_PATH, so this workaround is needed
+          cat <(echo "export PATH=$PATH:$(pwd)/.github/bin") ~/.bashrc > ~/.bashrc.new
+          mv ~/.bashrc.new ~/.bashrc
+        else
+          # MacOS and Linux use $GITUB_PATH
+          echo "$(pwd)/.github/bin" >> $GITHUB_PATH
+        fi
 
-    - name: Install build environment
-      if: matrix.os == 'ubuntu-latest'
+    - name: "WIN: Install build environment (secp256k1)"
```

Comment:
The steps for installing secp256k1 are almost the same for every OS, could they be combined into one step?

Reply:
I combined the three of them in one step. Lets see if CI complains.

Thanks!